File: Sesion 6/Marco teorico/8CÓMO DECLARAR AMIGA PARCIALMENTE A UNA CLASE/8.java
class ClaseA {
    private int datoPrivadoA;

    public ClaseA(int dato) {
        this.datoPrivadoA = dato;
    }

    public int getDatoPrivadoA() {
        return datoPrivadoA;
    }
}

class ClaseB {
    public void metodoB(ClaseA objetoA) {
        System.out.println("El dato privado de ClaseA es: " + objetoA.getDatoPrivadoA());
    }
}

public class Main {
    public static void main(String[] args) {
        ClaseA objetoA = new ClaseA(42);
        ClaseB objetoB = new ClaseB();
        objetoB.metodoB(objetoA);
    }
}

File: Sesion 6/Marco teorico/8CÓMO DECLARAR AMIGA PARCIALMENTE A UNA CLASE/8.py
class ClaseA:
    def __init__(self, dato):
        self.__dato_privado_a = dato

class ClaseB:
    def metodo_b(self, objeto_a):
        print("El dato privado de ClaseA es:", objeto_a._ClaseA__dato_privado_a)


objeto_a = ClaseA(42)
objeto_b = ClaseB()
objeto_b.metodo_b(objeto_a)